ctapipe/irf/optimise.py

```diff
@@ -0,0 +1,149 @@
+import operator
+
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable, Table
+from pyirf.binning import create_bins_per_decade
+from pyirf.cut_optimization import optimize_gh_cut
+from pyirf.cuts import calculate_percentile_cut, evaluate_binned_cut
+
+from ..core import Component, QualityQuery
+from ..core.traits import Float
+
+
+class OptimisationResult:
+    def __init__(self, gh_cuts=None, offset_lim=None):
+        self.gh_cuts = gh_cuts
+        if gh_cuts:
+            self.gh_cuts.meta["extname"] = "GH_CUTS"
```

Comment:
extname is the name of a FITS extension. You should't set this in the metadata of the table, but only when writing out, it's the `name` argument of `write`.

Reply:
I got
```
TypeError: write_table_fits() got an unexpected keyword argument 'name'
```
when I tried `self.gh_cuts.write(out_name, name="GH_CUTS", format="fits", overwrite=overwrite)` and I don't find anything about a `name` argument to `table.write`, do you mean I should manage the fits file more directly and just cast the tables to hdus?